File: presentation/src/main/java/com/github/anrimian/musicplayer/ui/library/albums/items/AlbumItemsView.java
package com.github.anrimian.musicplayer.ui.library.albums.items;

import com.github.anrimian.musicplayer.domain.models.albums.Album;
import com.github.anrimian.musicplayer.ui.library.common.compositions.BaseLibraryCompositionsView;

import moxy.viewstate.strategy.alias.AddToEndSingle;
import moxy.viewstate.strategy.alias.OneExecution;
import moxy.viewstate.strategy.alias.Skip;

public interface AlbumItemsView extends BaseLibraryCompositionsView {

    @AddToEndSingle
    void showAlbumInfo(Album album);

    @OneExecution
    void closeScreen();

    @Skip
    void showEditAlbumScreen(Album album);
}

File: app/src/main/java/com/github/anrimian/musicplayer/ui/library/albums/items/AlbumItemsView.kt
package com.github.anrimian.musicplayer.ui.library.albums.items

import com.github.anrimian.musicplayer.domain.models.albums.Album
import com.github.anrimian.musicplayer.ui.library.common.compositions.BaseLibraryCompositionsView
import moxy.viewstate.strategy.alias.AddToEndSingle
import moxy.viewstate.strategy.alias.OneExecution
import moxy.viewstate.strategy.alias.Skip

interface AlbumItemsView : BaseLibraryCompositionsView {

    @AddToEndSingle
    fun showAlbumInfo(album: Album)

    @OneExecution
    fun closeScreen()

    @Skip
    fun showEditAlbumScreen(album: Album)

}